### PR TITLE
fix: update button states to secondary

### PR DESCRIPTION
**Pull request by tanvibhakta**
Some tertiary buttons were missed out when closing https://github.com/appsmithorg/appsmith/issues/18052, not sure why. This PR updates them.

Depends on https://github.com/appsmithorg/design-system/pull/294

### CODE REVIEW

**Review by ankitakinger — APPROVED** [review]

**Comment by shadabbuchh** [comment]
Issues found:

1. The `Save/Edit Datasource` button does not look like any button. 
Screenshot 2022-12-23 at 3 33 38 PM
2. Tertiary button is being used for Snippets
Screenshot 2022-12-23 at 3 34 29 PM

**Comment by tanvibhakta** [comment]
1. It looks like the `Save/Edit datasource` button has not been using the design-system button for a while. 
2. Snippet button has been updated.

File: app/client/package.json [hunk 1]
```diff
@@ -46,7 +46,7 @@
     "cypress-log-to-output": "^1.1.2",
     "dayjs": "^1.10.6",
     "deep-diff": "^1.0.2",
-    "design-system": "npm:@appsmithorg/design-system@1.0.41",
+    "design-system": "npm:@appsmithorg/design-system@1.0.42-alpha.2",
```
Comment (ankitakinger) [inline app/client/package.json:49]: is the version here intentional?

Comment (tanvibhakta) [reply app/client/package.json:49]: Yes, it needs to be the alpha version in order to use the latest update

**Comment by shadabbuchh** [comment]
`Snippets` is now utilising the secondary button.

**Review by ankitakinger — APPROVED** [review]

File: app/client/package.json (hunk 1 above)
Comment (tanvibhakta) [reply app/client/package.json:49]: yes - this needs to be an alpha version in order to use the latest design system changes

**Review by ankitakinger — APPROVED** [review]

**Review by ankitakinger — APPROVED** [review]

**Pull request merged**